src/Stratis.CirrusD/Program.cs

```diff
@@ -56,6 +57,7 @@ private static IFullNode GetSideChainFullNode(NodeSettings nodeSettings)
                 .UseSmartContractWallet()
                 .UseApi()
                 .AddRPC()
+                .UseDiagnosticFeature()
```

Comment:
Probably not by default, here I am not sure how to proceeded.
If we add diagnostics then it should not be as a feature but a built in part of the node (basically add the components to base) no?

Reply:
> Probably not by default, here I am not sure how to proceeded.

note that currently I've implemented in such a way that if you don't run the node with the `-diagpeerstats` argument you do not run any code related to that feature (it just registers the controller that let you know start diagnostic at will and obtain stats) so running it by default its fine.
Only with that argument or running `StartCollectingPeerStatistics` will register to these events and will start an async loop to collect and store them in memory.

> If we add diagnostics then it should not be as a feature but a built in part of the node (basically add the components to base) no?

Then you could say the same thing about wallet, api, rpc or smartcontract.
I even imagine we'll add other diagnostic stuff in this feature in future.
Technically you may want to not include a diagnosticfeature in a node you consider stable, so you are free to not include it in your final assembly (less code, less memory taken by the process, Raspberry Pi will thank you)

Microsoft too decided to create little assemblies for .net framework (e.g. look at how many little assemblies you refer working on .netcore)

Anyway I'm not totally against moving this statistic collector inside base feature, even if I don't see any cons about having its own feature but only pros, feel free to convince me of the opposite :)

Reply:
>I've implemented in such a way that if you don't run the node with the -diagpeerstats argument 
  
Well we will still create instances of events that no one listens to.

>Then you could say the same thing about wallet, api, rpc or smartcontract.  

No I disagree those are products of the fullnode diagnostics is a tool more like configuration or logging.

Reply:
> Well we will still create instances of events that no one listens to.
yes, agree, like every logger.Trace/Debug do passing strings and arguments (but called a number of times with higher orders of magnitude)

without considering all the times we capture contexts using lambdas, that generate the instantiations of temporary objects moved between contexts, etc...

We should care of these kind of issues only backed by numbers.
Event Bus works this way on enterprise products, there could be micro optimization at a cost of usabillity or design complexity that actually aren't worth to be taken into account imo.